File: LeetCode/Easy/Ruby/best_time_stock_sale.rb
def max_profit(prices)
  if prices.size < 2 return 0
  else 
    max_profit = 0
    min_price = prices[0]
    (1..prices.count-1).each do |i|
      min_price = prices|i| if prices|i| < min_price
      max_profit = prices[i] - min_price if prices[i] - min_price > max_profit
    end
  return max_profit
  end
end

File: LeetCode/Easy/JavaScript/bestTimeStockSale.js
const maxProfit = prices => {
    if (prices.length < 2) return 0
    else {
    let profit = 0;
    let lowestPrice  = prices[0];
    for(let i = 1; i < prices.length; i++) {
        lowestPrice = Math.min(prices[i], lowestPrice);
        profit = Math.max(profit, prices[i] - lowestPrice);
    }
    return profit;
    }
};
